app/main.py

```diff
@@ -1,34 +1,110 @@
+from __future__ import annotations
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: int, end: int, is_drowned: bool = False) -> None:
```

Comment:
The `start` and `end` parameters are expected to be tuples representing coordinates, but they are being used as if they are iterable with indices. Ensure that `start` and `end` are tuples of integers, e.g., `(row, column)`.